File: tests/07_phases_ompss.dg/fortran/success_task_28.f90
! <testinfo>
! test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
! test_nolink=yes
! </testinfo>

SUBROUTINE FOO(N)
    IMPLICIT NONE
    INTEGER :: N
    INTEGER :: A(N)
    INTEGER :: B(N)

    !$OMP TASK SHARED(A) INOUT(B)
    !$OMP END TASK

    !$OMP TASKWAIT
END SUBROUTINE FOO

File: tests/07_phases_ompss.dg/c/success_task_28.c
/*
<testinfo>
test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
test_nolink=yes
</testinfo>
*/
int foo(int *v, int m, int n) {
    #pragma omp task inout( ((int (*)[n])v)[0;m])
    {}
}
